lib/wasi/src/runners/wasi.rs

```diff
@@ -95,7 +95,12 @@ impl crate::runners::Runner for WasiRunner {
             builder.set_runtime(Arc::new(rt));
         }
 
-        builder.run(module)?;
+        let res = builder.run(module);
+        match res {
+            Ok(()) => Ok(()),
+            Err(crate::WasiRuntimeError::Wasi(crate::WasiError::Exit(_))) => Ok(()),
```

Comment:
This means any exit code will indicate a success. What about modifying it to only report an exit code of `0` as `Ok(())`?

```suggestion
            Err(crate::WasiRuntimeError::Wasi(crate::WasiError::Exit(0))) => Ok(()),
```

Reply:
After talking with @ptitSeb, we don't want to make assumptions about the exit code so unconditionally returning `Ok(())` is fine.

Reply:
Exit Code !=0 are still normal exit code. Will not make any assumption on the exit code.